### PR TITLE
feat: add build github action, remove unused env vars

### PR DESCRIPTION
**Pull request by MantisClone**
# Problem

CI doesn't verify that code builds successfully. This is the most basic smoke test for an app and is trivial to implement.

# Changes

* feat: add build github action
* fix: remove unused env vars

## Summary by CodeRabbit

- **New Features**
	- Introduced a new deployment workflow to streamline the process of deploying to production.
	- The workflow can be triggered manually or by pull requests.
- **Chores**
	- Added steps for checking out the code, setting up Node.js, installing dependencies, and building the application.
	- Removed unnecessary environment variables from the `.env.example` file to simplify configuration.
- **Documentation**
	- Updated the README.md to reflect changes in the deployment process by removing certain environment variables for Vercel.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Deploy to production
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [opened, reopened]
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply .github/workflows/build.yml]: 1. Fixed :white_check_mark: 
2. Fixed :white_check_mark: 
3. Unnecessary. No Problem :ok_person:

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Build
+        run: npm run build
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply .github/workflows/build.yml:19]: 1. Fixed :white_check_mark: 
2. Fixed :white_check_mark: 
3. Timeout not needed. No Problem :ok_person: 
4. Artifacts not necessary. No Problem :ok_person:

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: Build
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [opened, reopened]
+
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply .github/workflows/build.yml]: 1. Fixed :white_check_mark: 
2. Fixed :white_check_mark:

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Build
+        run: npm run build
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply .github/workflows/build.yml:33]: No Problem :ok_person:

**Comment by MantisClone** [comment]
@coderabbitai full review

**Review by rodrigopavezi — APPROVED** [review]

**Pull request merged**